### DISCUSSION TITLE
How to debug library when it doesn't work?

### DISCUSSION THREAD

**Discussion by cheryllium** [chimurai/http-proxy-middleware#685] [Q&A]
Today, the proxy stopped working in my development environment for a project I am working on. This suddenly started to happen after removing and recreating the docker container for the project.

The issue was reproducible. However, it is consistent. The problem occurs for every commit in the project history that I've checked, even commits that I'm 100% certain used to work in the past from a newly created docker container. 

No changes were made to any of the proxy code, the version of http-proxy-middleware and other dependencies, etc. The setup code is extremely basic, very similar to the example code provided by the documentation, and has worked just fine up until today's restart.

My question is: how do I debug this library when it doesn't work? I'd appreciate any tips: 

- Is there a way to make the error message show more verbose output? 
- Is there a way to use the Node debugger to step through what happens when it receives a request from the frontend?
- Does anyone have other suggestions to see what is going on inside the black box when problems occur?

I am just at a loss as to how to proceed with debugging this problem. I can't fathom what possibly introduced the problem. So far, evidence indicates it is not caused by any changes to the code, the environment, or the dependencies. 

I have considered it may be a problem with the docker container. It was (not very intelligently) based off of `node:lts` image, which had recent updates. But I have tried setting it to `node:16` through `node:12` with no change to this problem.

**Comment by chimurai** [comment]
You can try [v3@beta](https://github.com/chimurai/http-proxy-middleware/discussions/768).

You should be able to get better debug and log data.

```js
createProxyMiddleware({
   logger: console
})
```

```shell
# enable debug logging
DEBUG=http-proxy-middleware* node app.js
```
